contrib/TestHarness2/test_harness/joshua.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+import collections
+import io
+import sys
+import xml.sax
+import xml.sax.handler
+from pathlib import Path
+from typing import List, OrderedDict, Set
+
+from joshua import joshua_model
+
+import test_harness.run
+from test_harness.config import config
+from test_harness.summarize import SummaryTree
+
+
+class ToSummaryTree(xml.sax.handler.ContentHandler):
+    def __init__(self):
+        super().__init__()
+        self.root: SummaryTree | None = None
+        self.stack: List[SummaryTree] = []
+
+    def result(self) -> SummaryTree:
+        assert len(self.stack) == 0 and self.root is not None, 'Parse Error'
+        return self.root
+
+    def startElement(self, name, attrs):
+        new_child = SummaryTree(name)
+        for k, v in attrs.items():
+            new_child.attributes[k] = v
+        self.stack.append(new_child)
+
+    def endElement(self, name):
+        closed = self.stack.pop()
+        assert closed.name == name
+        if len(self.stack) == 0:
+            self.root = closed
+        else:
+            self.stack[-1].children.append(closed)
+
+
+def _print_summary(summary: SummaryTree, commands: Set[str]):
+    cmd = []
+    if config.reproduce_prefix is not None:
+        cmd.append(config.reproduce_prefix)
+    cmd.append('fdbserver')
+    if 'TestFile' in summary.attributes:
+        file_name = summary.attributes['TestFile']
+        role = 'test' if test_harness.run.is_no_sim(Path(file_name)) else 'simulation'
+        cmd += ['-r', role, '-f', file_name]
+    else:
+        cmd += ['-r', 'simulation', '-f', '<ERROR>']
+    if 'RandomSeed' in summary.attributes:
+        cmd += ['-s', summary.attributes['RandomSeed']]
+    else:
+        cmd += ['-s', '<Error>']
+    if 'BuggifyEnabled' in summary.attributes:
+        arg = 'on'
+        if summary.attributes['BuggifyEnabled'].lower() in ['0', 'off', 'false']:
+            arg = 'off'
+        cmd += ['-b', arg]
+    else:
+        cmd += ['b', '<ERROR>']
+    cmd += ['--crash', '--trace_format', 'json']
```

Comment:
The reason I hard-coded this was because this is meant to replace `devJoshuaResults` which also hard-codes this. But I think you're right and my reasoning was wrong. I will change this.